In [2]:
from datascience import Table
import pandas as pd

# Load dataset into a datascience Table
fastfood_table = Table.read_table('Fastfood_USA.csv')


In [ ]:
# Convert the Table to a Pandas DataFrame
fastfood_df = fastfood_table.to_df()

# Check for missing values using isnull() in Pandas DataFrame
missing_values = fastfood_df.isnull().sum()
print(missing_values)

In [ ]:
fastfood_table.columns

In [ ]:
fastfood_df.head(10)  # Displays the first 10 rows of the DataFrame


In [ ]:
fastfood_df.tail(10)  # Displays the last 10 rows of the DataFrame

In [ ]:
# @title Which fast-food chain had the highest systemwide sales?
# Find the fast-food chain with the highest systemwide sales in 2021
highest_sales_chain = fastfood_df.loc[fastfood_df['U.S. Systemwide Sales (Millions - U.S Dollars)'].idxmax()]

# Display the fast-food chain with the highest systemwide sales in 2021
print("Fast-Food Chain with the Highest Systemwide Sales in 2021:")
print(highest_sales_chain[['Fast-Food Chains', 'U.S. Systemwide Sales (Millions - U.S Dollars)']])



The code utilizes the idxmax() function to find the index of the maximum value in the 'U.S. Systemwide Sales (Millions - U.S Dollars)' column within the DataFrame. It then employs the loc[] function to identify the row corresponding to this maximum value. This row information, which represents the fast-food chain with the highest systemwide sales in 2021, is stored in the variable highest_sales_chain. Finally, the print() function is used to display the name of this fast-food chain along with its associated sales value for the year 2021.

**McDonald's leads in total systemwide sales, amassing a significant sum of 45,960 million U.S. dollars in U.S. Systemwide Sales (Millions - U.S Dollars).**

In [ ]:
# @title List 10 fast-food chains with the least sales.
# Sort by U.S. Systemwide Sales to identify chains with least sales
bottom_sales_chains = fastfood_table.sort('U.S. Systemwide Sales (Millions - U.S Dollars)').take[:10]

print("Chains with the least U.S. Systemwide Sales:")
print(bottom_sales_chains.select(['Fast-Food Chains', 'U.S. Systemwide Sales (Millions - U.S Dollars)']))


This code snippet efficiently identifies and prints the ten fast-food chains with the lowest U.S. Systemwide Sales. It first sorts the dataset by the 'U.S. Systemwide Sales (Millions - U.S Dollars)' column in ascending order, revealing the chains with the lowest sales figures. Then, it selects the first ten entries from this sorted list to display the names of these chains along with their respective U.S. Systemwide Sales values.

In [ ]:
# @title Identify the top 10 performing chains in terms of U.S. Systemwide Sales?
top_sales_chains = fastfood_table.sort('U.S. Systemwide Sales (Millions - U.S Dollars)', descending=True).take[:10]

print("Top-performing chains based on U.S. Systemwide Sales:")
print(top_sales_chains.select(['Fast-Food Chains', 'U.S. Systemwide Sales (Millions - U.S Dollars)']))



This code snippet sorts the dataset in descending order using the 'U.S. Systemwide Sales (Millions - U.S Dollars)' column. By doing so, it arranges the data from highest to lowest sales figures. Then, it selects the first ten entries from this sorted list, containing the names of the top-performing chains along with their respective U.S. Systemwide Sales values.

In [ ]:
# @title Identify the 10 top-performing chains in terms of Average Sales per Unit?

# Sort by Average Sales per Unit to identify top-performing chains
top_avg_sales_chains = fastfood_table.sort('Average Sales per Unit (Thousands - U.S Dollars)', descending=True).take[:10]

print("\nTop-performing chains based on Average Sales per Unit:")
print(top_avg_sales_chains.select(['Fast-Food Chains', 'Average Sales per Unit (Thousands - U.S Dollars)']))

This code snippet is intended to identify and display the top 10 performing fast-food chains based on their Average Sales per Unit. The process involves sorting the dataset in descending order using the 'Average Sales per Unit (Thousands - U.S Dollars)' column. This sorting operation arranges the data from highest to lowest values of average sales per unit. Then, it selects the first ten entries from this sorted list, which contains the names of the top-performing chains along with their corresponding Average Sales per Unit values.

In [ ]:
# @title Is there a correlation between the U.S. Systemwide Sales values and the Average Sales per Unit?
import pandas as pd

# Convert the Table to a Pandas DataFrame
fastfood_df = fastfood_table.to_df()

# Calculate correlation between U.S. Systemwide Sales and Average Sales per Unit
sales_correlation = fastfood_df['U.S. Systemwide Sales (Millions - U.S Dollars)'].corr(
    fastfood_df['Average Sales per Unit (Thousands - U.S Dollars)']
)

print("Correlation between U.S. Systemwide Sales and Average Sales per Unit:", sales_correlation)


This snippet converts the data from the datascience Table format to a Pandas DataFrame using the to_df() method. It employs the corr() function to compute the correlation between the 'U.S. Systemwide Sales (Millions - U.S Dollars)' column and the 'Average Sales per Unit (Thousands - U.S Dollars)' column in the Pandas DataFrame.
The resulting value, stored in the variable sales_correlation, represents the correlation coefficient between these two variables. Finally, the code prints out the computed correlation coefficient using the print() function.

 **A correlation coefficient of 0.231 indicates a relatively weak positive linear relationship between U.S. Systemwide Sales and Average Sales per Unit in the fast-food chains dataset. This positive correlation suggests that there is some tendency for higher U.S. Systemwide Sales to be associated with higher Average Sales per Unit, but the relationship is not very strong.**

In [ ]:
# @title Which fast-food chain achieved the highest average sales per unit while maintaining a low number of franchised stores?
# Filter for chains with a low number of franchised stores (e.g., less than 50)
low_franchise_threshold = 50
low_franchise_chains = fastfood_df[fastfood_df['Franchised Stores'] < low_franchise_threshold].copy()

# Calculate average sales per unit for these chains
low_franchise_chains['Sales per Unit'] = low_franchise_chains['U.S. Systemwide Sales (Millions - U.S Dollars)'] / low_franchise_chains['2021 Total Units']

# Find the chain with the highest average sales per unit among these low franchised stores chains
highest_avg_sales_chain = low_franchise_chains.loc[low_franchise_chains['Sales per Unit'].idxmax()]

# Display the chain with the highest average sales per unit and low franchised stores
print("Chain with the highest average sales per unit and low franchised stores:")
print(highest_avg_sales_chain[['Fast-Food Chains', 'Average Sales per Unit (Thousands - U.S Dollars)', 'Franchised Stores']])


This snippet of code filters fast-food chains with a low number of franchised stores (less than 50) from the given dataset. Then, it calculates the average sales per unit for these filtered chains by dividing their total systemwide sales by the total number of units in 2021. Next, it identifies the chain among these filtered ones that has the highest average sales per unit. Finally, it displays information about this specific chain, including its name, average sales per unit, and the number of franchised stores it operates. The .copy() method ensures a separate copy is created to avoid certain warnings when working with dataframes.


**Raising Cane’s achieves an average sales per unit of $4893 thousand. Raising Cane’s has the highest average sales per unit among chains with a low number of franchised stores. The chain operates only 23 franchised stores while maintaining high average sales per unit, indicating a relatively high profitability per store.**

In [19]:
# @title What is the total number of fast-food franchises open in the U.S?
# Calculate the total number of fast-food franchises in the U.S.
total_franchises = fastfood_table['Franchised Stores'].sum()
print("Total number of fast-food franchises open in the U.S.: ", total_franchises)


Total number of fast-food franchises open in the U.S.:  133198


This snippet of code fetches the column 'Franchised Stores' from the 'fastfood_table' DataFrame, which represents the number of franchised stores for each fast-food chain. The sum() function is applied to the 'Franchised Stores' column, which adds up all the values in that column. The result is stored in the variable total_franchises, representing the total count of franchised stores across all fast-food chains. The print() statement outputs the total number of fast-food franchises open in the U.S., which is stored in the variable total_franchises.

**The output indicates that the current total number of fast-food franchises open in the United States is 133,198.**


In [ ]:
# @title Which fast food chain experienced the most growth from 2020 to 2021?
# Find the fast-food chain with the most growth from 2020 to 2021
most_growth_chain = fastfood_df.loc[fastfood_df['Total Change in Units from 2020'].idxmax()]
print("Fast-food chain with the most growth from 2020 to 2021:")
print(most_growth_chain[['Fast-Food Chains', 'Total Change in Units from 2020']])



This code snippet identifies the fast-food chain that experienced the most growth from 2020 to 2021 in terms of the total change in units. most_growth_chain finds the row of data for the fast-food chain with the highest increase in the total number of units from 2020 to 2021.
fastfood_df['Total Change in Units from 2020'].idxmax() finds the index (row number) of the maximum value in the 'Total Change in Units from 2020' column, representing the chain that experienced the most growth.
most_growth_chain[['Fast-Food Chains', 'Total Change in Units from 2020']] displays the name of the fast-food chain and the total change in units it experienced between 2020 and 2021.

**The output provided displays information indicating that the fast-food chain, Jersey Mike’s experienced the most growth in the number of units among all fast-food chains during this period with a total increase of 246 units from 2020 to 2021.**

In [17]:
# @title Identify the top 3 performing chains in terms of U.S. Systemwide Sales Growth Rate from 2020 to 2021?
# Calculate growth rate
fastfood_df['Growth Rate'] = (fastfood_df['U.S. Systemwide Sales (Millions - U.S Dollars)'] / (fastfood_df['Total Change in Units from 2020'] + 1)) - 1

# Get the top 3 performing chains in terms of growth rate
top_3_growth_chains = fastfood_df.nlargest(3, 'Growth Rate')[['Fast-Food Chains', 'Growth Rate']]

print("Top 3 performing chains in terms of U.S. Systemwide Sales Growth Rate from 2020 to 2021:")
print(top_3_growth_chains)


Top 3 performing chains in terms of U.S. Systemwide Sales Growth Rate from 2020 to 2021:
   Fast-Food Chains  Growth Rate
49          Zaxby’s   557.250000
14    El Pollo Loco   485.500000
23              KFC   462.636364


The code calculates the growth rate for each fast-food chain using the formula:
Growth Rate = (U.S. Systemwide Sales / (Total Change in Units from 2020 + 1)) - 1
It computes the growth rate by dividing the U.S. Systemwide Sales by the total change in units from 2020 and subtracts 1 to represent the percentage change.
The nlargest() function is used to obtain the top 3 performing chains based on the 'Growth Rate' column, which represents the highest growth rates. The code selects the columns 'Fast-Food Chains' and 'Growth Rate' for the top 3 chains.

**Zaxby’s: Experienced a Growth Rate of 557.25%**

**El Pollo Loco: Achieved a Growth Rate of 485.5%**

**KFC: Recorded a Growth Rate of 462.64%**

## CONCLUSION
####The analysis of the top 50 fast-food chains in the US is crucial for understanding market dynamics, identifying industry leaders, and assessing sales trends. It provides insights into profitability, franchise models, and growth potential, offering valuable information for strategic decision-making and market positioning within the competitive fast-food industry.##